tests/ssg_test_suite/rule.py

```diff
@@ -399,30 +405,32 @@ def _get_rule_scenarios(self, rule):
                 templated_test_scenarios.keys())
         all_tests.update(templated_test_scenarios)
         all_tests.update(local_test_scenarios)
+        return all_tests
 
-        # Filter out everything except the shell test scenarios.
-        # Other files in rule directories are editor swap files
-        # or other content than a test case.
-        allowed_scripts = filter(lambda x: x.endswith(".sh"), all_tests)
-        content_mapping = {x: all_tests[x] for x in allowed_scripts}
-
+    def _get_rule_test_content(self, rule):
```

Comment:
CodeClimate is complaining about complexity of the function. If you don't see any meaningful way how to reduce it except refactoring the "if with and", then please also modify the codeclimate yaml to soften its settings accordingly. That coplexity would be somehow acceptable.